offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -517,15 +529,27 @@ class Offer {
     }
 
     private double calculateTotalCosts(){
-        final double netPrice = calculateNetPrice()
+        double netPrice = calculateNetPrice()
         final double overhead = getOverheadSum()
-        return netPrice + overhead + getTaxCosts() - getTotalDiscountAmount()
+        return netPrice + overhead + getTaxCosts()
     }
 
     private ProductItem finaliseProductItem(ProductItem item) {
-        BigDecimal totalItemCosts = calculateItemNet(item)
         BigDecimal totalItemQuantityDiscount = discountAmountForProductItem(item)
```

Comment:
Maybe you should rename the `discountAmountForProductItem` method to `quantityDiscountForProductItem` or something like this, since you introduce a new kind of discount.